### ISSUE TITLE
Is it possible to use flake-compat without flakes enabled?

### ISSUE THREAD

**Issue by jalaziz** [edolstra/flake-compat#66]
My understanding was that `flake-compat` could be used to enable the use of flakes with stable Nix. However, when I run `nix-shell` for my devShell based project, I receive:

```
error: experimental Nix feature 'flakes' is disabled; add '--extra-experimental-features flakes' to enable it
```

Am I misunderstanding the purpose of `flake-compat`?

**Comment by justinas** [comment]
The purpose is to allow you to import a flake *into* a non-flake Nix project.

> for my devShell based project,

If you're making your own `flake.nix`, and using flake-enabled commands such as `nix shell` (you mentioned `nix-shell`, but I'm not sure if you meant to?), you still need to enable the flakes experimental feature.

**Comment by edolstra** [comment]
> The purpose is to allow you to import a flake into a non-flake Nix project.

That's right. It shouldn't be necessary to enable the `flakes` experimental feature.

> The purpose is to allow you to import a flake into a non-flake Nix project.

No, that's what `builtins.getFlake` is for.
